File: AoC_2022/day01/script01.py
with open('data01.txt') as d:
    data = [el.strip() for el in d.readlines()]

data_dict = {}
elf = 1
for d in data:
    if d == "":
        elf += 1
    else:
        data_dict[elf] = data_dict.get(elf, 0) + int(d)

print("Star 1 =", max(data_dict.values()))
print("Star 2 =" ,sum(sorted(data_dict.values())[-3:]))

File: AoC_2022/day01/script01.R
library(tidyverse)
ints <- as.numeric(readLines("data01.txt"))

cals <- tibble(ints = ints) |> 
  mutate(elve = cumsum(is.na(ints))) |> 
  group_by(elve) |> 
  summarise(tot = sum(ints, na.rm = TRUE)) |> 
  arrange(desc(tot)) |> 
  pull()

paste("Star 1 =", cals[1])
paste("Star 2 =", sum(cals[1:3]))
